src/components/MultiHopTrade/hooks/useSupportedAssets.tsx

```diff
@@ -6,23 +6,31 @@ import { walletSupportsChain } from 'hooks/useWalletSupportsChain/useWalletSuppo
 import { isSome } from 'lib/utils'
 import { useGetSupportedAssetsQuery } from 'state/apis/swapper/swapperApi'
 import { selectAssetsSortedByMarketCapUserCurrencyBalanceAndName } from 'state/slices/common-selectors'
-import { selectAssets } from 'state/slices/selectors'
-import { useAppSelector } from 'state/store'
+import {
+  selectAccountIdsByChainId,
+  selectAssets,
+  selectPortfolioAccounts,
+} from 'state/slices/selectors'
+import { store, useAppSelector } from 'state/store'
 
 export const useSupportedAssets = () => {
   const sortedAssets = useAppSelector(selectAssetsSortedByMarketCapUserCurrencyBalanceAndName)
   const assets = useAppSelector(selectAssets)
   const wallet = useWallet().state.wallet
   const isSnapInstalled = useIsSnapInstalled()
 
+  const portfolioAccounts = useAppSelector(selectPortfolioAccounts)
```

Comment:
I would rename `selectAccountIdsByChainId` -> `selectAccountIdsByChainIdFilter` and add a new selector `selectAccountIdsByChainId` that returns the whole map of `chainId` -> `accountIds`. Then we can use this value in the dependency arrays without the need for the current react on `portfolioAccounts` to trigger the raw `store.getState`

Reply:
1. Nice one - indeed the naming would imply this is a mapping of `ChainId` to `AccountId[]` rather than an `AccountId[]` list. 393a151d4e6b730cc5db86ebe75f2c28f1005596